tests/bats/cleanup-from-previous-run.sh

```diff
@@ -62,15 +62,24 @@ timeout -v 5 kubectl delete pods -l env=batssuite 2> /dev/null
 timeout -v 2 kubectl delete resourceclaim batssuite-rc-bad-opaque-config --force 2> /dev/null
 timeout -v 2 kubectl delete -f demo/specs/imex/simple-mig-test 2> /dev/null
 
+# Cleanup any GPU stress test pods left behind
+timeout -v 30 kubectl delete pods -l 'env=batssuite,test=stress-shared' 2> /dev/null
+timeout -v 5 kubectl delete -f tests/bats/specs/rc-shared-gpu.yaml 2> /dev/null
+kubectl wait --for=delete pods -l 'env=batssuite,test=stress-shared' \
+    --timeout=60s \
+    || echo "wait-for-delete failed"
+
 # TODO: maybe more brute-forcing/best-effort: it might make sense to submit all
 # workload in this test suite into a special namespace (not `default`), and to
 # then use `kubectl delete pods -n <testnamespace]> --all`.
 
 # Delete any previous remainder of `clean-state-dirs-all-nodes.sh` invocation.
 kubectl delete pods privpod-rm-plugindirs 2> /dev/null
 
-timeout -v 5 helm uninstall nvidia-dra-driver-gpu-batssuite -n nvidia-dra-driver-gpu
+# Make sure to wait till the chart is completely removed
+helm uninstall nvidia-dra-driver-gpu-batssuite --wait -n nvidia-dra-driver-gpu
```

Comment:
👍 

That one is interesting. I'll see how it does in practice for me!

I've found Helm's `--wait` surprisingly surprising at times.